### PR TITLE
Feature online planned traj

### PR DESCRIPTION
**Pull request by Robert108**

### CODE REVIEW

File: include/grem.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    uint16_t ReceivedHeaderMessageCounterContentLength;
+    uint8_t ReceivedHeaderMessageCounter;
+
+    uint16_t ReceivedHeaderMessageIDValueID;
+    uint16_t ReceivedHeaderMessageIDContentLength;
+    uint16_t ReceivedHeaderMessageID;
+
+    uint16_t ResponseCodeValueID;
+    uint16_t ResponseCodeContentLength;
+	uint8_t ResponseCode;
```
Comment (konglobemeralt) [inline include/grem.h:26]:
```suggestion
    uint8_t ResponseCode;
```

File: include/grem.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+#include "iso22133.h"
+#include "header.h"
+#include "footer.h"
+
+#pragma pack(push, 1)
+typedef struct{
+	HeaderType header;
+    uint16_t ReceivedHeaderTransmitterIDValueID;
+	uint16_t ReceivedHeaderTransmitterIDContentLength;
```
Comment (konglobemeralt) [inline include/grem.h:13]:
```suggestion
    uint16_t ReceivedHeaderTransmitterIDContentLength;
```

File: include/grem.h [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+#include <stdint.h>
+
+#include "iso22133.h"
+#include "header.h"
+#include "footer.h"
+
+#pragma pack(push, 1)
+typedef struct{
+	HeaderType header;
```
Comment (konglobemeralt) [inline include/grem.h:11]:
```suggestion
    HeaderType header;
```

File: include/grem.h [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    uint16_t ResponseCodeContentLength;
+	uint8_t ResponseCode;
+
+    uint16_t PayloadLengthValueID;
+    uint16_t PayloadLengthContentLength;
+    uint16_t PayloadLength;
+
+    uint16_t PayloadDataValueID;
+    uint16_t PayloadDataContentLength;
+	FooterType footer;
```
Comment (konglobemeralt) [inline include/grem.h:34]:
```suggestion
    FooterType footer;
```

File: iso22133.h [hunk 5]
```diff
@@ -314,13 +327,15 @@ enum ISOUnitType {
...
-	GREM_WRONG_ID = 5,
-	GREM_OUT_OF_RANGE = 6
-};
+    UNDEFIEND_0 = 0,
+	GREM_OK = 1,
+	GREM_GENERAL_ERROR = 2,
+	GREM_NOT_SUPPORTED = 3,
+	GREM_NO_MEMORY = 4,
+	GREM_INVALID_DATA = 5,
+	GREM_CHUNK_RECEIVED = 6,
```
Comment (konglobemeralt) [inline iso22133.h:336]: Nitpicky but spaces should be used instead of tabs.

File: iso22133.h [hunk 6]
```diff
@@ -460,7 +475,12 @@ typedef struct {
 } StartMessageType;
 
 typedef struct{
-	enum GeneralResponseStatus status;
+    uint32_t receivedHeaderTransmitterID;
+    uint8_t receivedHeaderMessageCounter;
+    uint16_t receivedHeaderMessageID; 
+	uint8_t responseCode;
```
Comment (konglobemeralt) [inline iso22133.h:481]:
```suggestion
    uint8_t responseCode;
```

**Review by konglobemeralt — COMMENTED** [review]
I general this looks good. I would suggest that @sepast takes a look as well with perhaps some more focus on the ISO specifics, but other than that I mostly commented on some formatting issues.

File: include/header.h
Comment (sepast) [inline include/header.h:1]: This content is needed! Where is it now?

**Review by sepast — COMMENTED** [review]
Some comments

File: iso22133.c [hunk 7]
```diff
@@ -610,6 +596,30 @@ enum ISOMessageID getISOMessageType(const char *messageData, const size_t length
 	}
 }
 
+/*!
```
Comment (sepast) [inline iso22133.c:599]: Delete this

File: iso22133.h [hunk 8]
```diff
@@ -314,13 +314,15 @@ enum ISOUnitType {
...
-};
+    UNDEFIEND_0 = 0,
+	GREM_OK = 1,
+	GREM_GENERAL_ERROR = 2,
+	GREM_NOT_SUPPORTED = 3,
+	GREM_NO_MEMORY = 4,
+	GREM_INVALID_DATA = 5,
+	GREM_CHUNK_RECEIVED = 6,
+    UNDEFIEND_1 = 7
+} __attribute__ ((__packed__));
```
Comment (sepast) [inline iso22133.h:325]: Is this needed? Isn't pragma used for all files?

**Review by sepast — COMMENTED** [review]
We need to check PRAGMA before merging.

**Review by sepast — APPROVED** [review]
Ok!

**Pull request merged**